DifferenceGenerator/src/main/kotlin/DifferenceGenerator.kt

```diff
@@ -8,16 +8,19 @@ import java.awt.Color
 import java.awt.Graphics2D
 import java.awt.image.BufferedImage
 import java.io.File
+import javax.imageio.ImageIO
 
 class DifferenceGenerator(
     video1Path: String,
     video2Path: String,
     outputPath: String,
     private val algorithm: AlignmentAlgorithm<BufferedImage>,
-) : AbstractDifferenceGenerator(video1Path, video2Path, outputPath) {
+    maskPath: String? = null,
```

Comment:
In general, I am not a huge fan of having the argument as path and parsing the file inside the class. Seems like a case for factoring out the functionality into some extra class (eg PixelMask) and simply passing an instance of PixelMask to the DifferenceGenerator. Just thinking that as masks might not even come from a file, but from an action in the GUI

Reply:
interesting point but then again masks should be reusable and hence saveable somehow. but you are right we should refactor the whole path transfer

Reply:
Yes, I am not against the loading from file. Just not, that the DifferenceGenerator gets yet another task, that could be abstracted away from. The class grows continuously and I think, we could start applying the Single Responsibility Principle a bit more :)